File: backend/src/main/java/com/backend/backend/service/Appointment/AppointmentService.java
package com.backend.backend.service.Appointment;

import java.util.List;

import com.backend.backend.model.Appointment.Appointment;

public interface AppointmentService {

    Appointment add(Appointment appointment);

    Appointment getById(Integer id);

    List<Appointment> getAll();

    Appointment update(Appointment appointment);

    void delete(Integer id);
}

File: src/service/AppointmentService.js
import axios from 'axios';

const BASE_URL = 'http://localhost:8080/api/appointment';

export const AppointmentService = {

    async getAll() {
        const response = await axios.get(BASE_URL);
        return response.data;
    },

    async getById(id) {
        const response = await axios.get(`${BASE_URL}/${id}`);
        return response.data;
    },

    async add(appointment) {
        const response = await axios.post(BASE_URL, appointment);
        return response.data;
    },

    async update(appointment) {
        const response = await axios.put(BASE_URL, appointment);
        return response.data;
    },

    async delete(id) {
        const response = await axios.delete(`${BASE_URL}/${id}`);
        return response.data;
    }
};